Cgrid update box tests

daveh150 opened the pull request:
For detailed information about submitting Pull Requests (PRs) to the CICE-Consortium,
please refer to: <https://github.com/CICE-Consortium/About-Us/wiki/Resource-Index#information-for-developers>


## PR checklist
- [ ] Short (1 sentence) summary of your PR: 
    update gridsys box tests. Update ice_forcing for boxslotcyl CD grid test. 
- [ ] Developer(s): 
    David Hebert
- [ ] Suggest PR reviewers from list in the column to the right.
- [ ] Please copy the PR test results link or provide a summary of testing completed below.
    83 measured results of 83 total results
83 of 83 tests PASSED
0 of 83 tests PENDING
0 of 83 tests MISSING data
0 of 83 tests FAILED
 
- How much do the PR code changes differ from the unmodified code? 
    - [ ] bit for bit
    - [ ] different at roundoff level
    - [x ] more substantial 
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes
    - [x ] No
- Does this PR add any new test cases?
    - [x ] Yes
    - [ ] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [ ] Yes
    - [x ] No, does the documentation need to be updated at a later time?
        - [x ] Yes
        - [ ] No 
- [ ] Please provide any additional information or relevant details below:
Added method to test boxslotcyl on CD grid. Added 'boxislands' test.

apcraig committed:
split up atm and ocn forcing

daveh150 committed:
Changed restore_ice = false for symtests

daveh150 committed:
Merge branch 'boxes' of https://github.com/apcraig/CICE into boxtest

daveh150 committed:
Added more box tests to gridsys_suite.ts. Modified several set_nml for use with new islands grid and uniform forcing

daveh150 committed:
Fix for TPM. THis was also fixed by Dave bailey

daveh150 committed:
Merge boxtest into cgridDEV

daveh150 committed:
Added CD forcing to boxslotcyl forcing data

daveh150 committed:
ice_init.F90: moved grid_average_X2Y to be in proper place after set_state_var. ice_grid.F90: modified variable declaration in gridaverage_X2Y to use (:,:,:) for dimensions of passed arrays

daveh150 committed:
Bug fix in ice_grid.F90:get_box_kmt. Need to first initialize land mask to c1 before adding land locations

apcraig reviewed (changes requested):
grid_box_kmt seems like a pretty generic name for a pretty specific implementation.  how about grid_boxislands_kmt?  would need to change the call too.

the set_nml filenames should be all lower case and contain no special characters.  can we rename the set_nml.boxislands_n to set_nml.boxislandsn, same for _e and _ne.  change filenames and also names in the test suite (gridsys_suite.ts)

daveh150 committed:
ice_grid.F90 - renamed subroutine box_grid_kmt to grid_boxislands_kmt. Changed names of set_nml to remove underscores. Updated gridsys_suite to be consistent with set_nml.

daveh150 commented:
I changed grid_box_kmt to grid_boxislands_kmt to be more descriptive as Tony suggested. Also renamed the set_nml files to remove the '_' to help with how grid suite determines directory names.

daveh150 committed:
revert boxadv to have ktherm=0. Change gridsys_suite.ts options to run box test on multple PEs. Removed boxadv from gridsys_suite.ts

eclare108213 commented on cicecore/cicedynB/general/ice_forcing.F90:
```diff
@@ -648,15 +648,15 @@ subroutine get_forcing_atmo
+         call box2001_data_atm
       elseif (trim(atm_data_type) == 'uniform_northeast') then
-      ! dah: uniformm opotions inclued here to allow call to prepare_forcing
+      ! dah: uniformm options included here to allow call to prepare_forcing
```
This is fine, just remove the two comment lines to clean up

daveh150 replied:
It appears 'prepare_forcing' is not needed for the atm_data_type = 'uniform*'. We moved this section to the end of the if statement and commented the checks out in the event we find problems that need prepare_forcing.

eclare108213 commented on cicecore/cicedynB/general/ice_forcing.F90:
```diff
@@ -5372,18 +5385,67 @@ subroutine box2001_data
+
+      character(len=*), parameter :: subname = '(box2001_data_ocn)'
+
+      if (local_debug .and. my_task == master_task) write(nu_diag,*) subname,'fdbg start'
```
I think we should clean up lines like this when we've got this working

eclare108213 commented on cicecore/cicedynB/general/ice_init.F90:
```diff
@@ -2361,6 +2365,7 @@ subroutine init_state
                              vicen(:,:,  :,iblk), vsnon(:,:,  :,iblk), &
                              uvel (:,:,    iblk), vvel (:,:,    iblk))
 
+
```
extra empty line

daveh150 replied:
Done

eclare108213 commented on cicecore/cicedynB/general/ice_init.F90:
```diff
@@ -2372,6 +2377,32 @@ subroutine init_state
                         vicen, vsnon, &
                         ntrcr, trcrn)
 
+      ! set_state_var defines uvel/vel on U-Grid
```
remove both of these comments -- the first one is not generally true, only for boxslotcyl, and the second one is redundant

eclare108213 commented on cicecore/cicedynB/general/ice_init.F90:
```diff
@@ -2439,9 +2470,10 @@ subroutine set_state_var (nx_block, ny_block, &
       use ice_arrays_column, only: hin_max
       use ice_domain_size, only: nilyr, nslyr, nx_global, ny_global, ncat
-      use ice_grid, only: grid_type
+      use ice_grid, only: grid_type, grid_system, grid_average_X2Y
```
Are grid_system, grid_average_X2Y used here, or can these be removed?

daveh150 replied:
These were not used and removed

eclare108213 commented on cicecore/cicedynB/general/ice_init.F90:
```diff
@@ -2869,7 +2901,7 @@ subroutine boxslotcyl_data_vel(i,        j,       &
       use ice_domain_size, only: nx_global, ny_global
       use ice_grid, only: dxrect
-
+      
```
remove spaces (or tab) in this line

daveh150 replied:
Done

eclare108213 commented on configuration/scripts/options/set_nml.gbox80:
```diff
@@ -1,4 +1,5 @@
 ice_ic = 'default'
 grid_type = 'rectangular'
 atm_data_type = 'box2001'
+ocn_data_type = 'calm'
```
do we want atm_data_type etc in the gbox* option files?  since we have to specify these in addition to the 'full' options (box2001, boxslotcyl), and they might conflict?

daveh150 replied:
It seems that we do not need set_nml.gbox* at all. All of those options would be in the other set_nml (box2001, boxslotcyl, boxsym?), etc.

eclare108213 replied:
Those option files are used by the scripts, and might even be useful empty.  As discussed, we could keep grid_type in there and move the other flags.

daveh150 replied:
I left the ice_ic and grid_type in the gbox* for now. We should check if it is needed for other test suites.

daveh150 replied:
Left the gbox* as-is. We will address where to change the forcing updates in later PR.

eclare108213 reviewed (commented):
Only very minor comments except for the question about which namelist flags are in set_nml.gbox*

daveh150 committed:
Clean up blank lines and unused use statements. Removed check for uniform atm_data_type in get_forcing_atmo
